utils/sane-backends/Makefile

```diff
@@ -130,6 +130,20 @@ $(call Package/sane-backends/Default)
   USERID=saned:saned
 endef
 
+define Package/sane-daemon-avahi-dbus-service
+$(call Package/sane-backends/Default)
+  SECTION:=utils
+  CATEGORY:=Utilities
+  DEPENDS+=sane-daemon +avahi-dbus-daemon
+  TITLE+= (network daemon avahi service files)
+endef
+
+define Package/sane-daemon-avahi-nodbus-service
```

Comment:
On 16-03-24 11:52 PM, Luiz Angelo Daros de Luca wrote:
> In utils/sane-backends/Makefile
> <https://github.com/openwrt/packages/pull/2543#discussion_r57416881>:
>
>> @@ -130,6 +130,20 @@ $(call Package/sane-backends/Default)
>>    USERID=saned:saned
>>  endef
>>
>> +define Package/sane-daemon-avahi-dbus-service
>> +$(call Package/sane-backends/Default)
>> +  SECTION:=utils
>> +  CATEGORY:=Utilities
>> +  DEPENDS+=sane-daemon +avahi-dbus-daemon
>> +  TITLE+= (network daemon avahi service files)
>> +endef
>> +
>> +define Package/sane-daemon-avahi-nodbus-service
>
> As far as I can see, there is no difference between dbus and nodbus.
> Normally you have different packages when the binary is linked to
> different equivalent libraries (i.e. *ssl).
> If it is just a way to select different dependencies, it would be
> interesting to have this solved in DEPENDS.
> I guess both provides avahi-daemon.

The problem is that the PROVIDES mechanism only works for building not 
for opkg.  For opkg, when you have PROVIDES one of the package is 
*always* installed even if a different PROVIDES package is installed.

Basically PROVIDES is broken and I don't know how to fix it.

I have filed a ticket on this, but it's not likely a high priority item, 
or even been looked into.

Regards,

Daniel